app/src/backend/ECRFDownloadService.ts

```diff
@@ -0,0 +1,275 @@
+      );
+    }
+  }
+
+  private static transformDataForTemplate(
+    inventoryValues: InventoryValueWithActivityValues[],
+    inventoryYear: number,
+    t: any,
+  ): Record<string, any> {
+    // Get the translation
+
+    const dataDictionary: Record<string, any> = {};
+    inventoryValues.forEach((inventoryValue) => {
+      const gpcRefNo = inventoryValue.gpcReferenceNumber;
+      const activityValues = inventoryValue.activityValues || [];
+
+      let methodologyDescription = inventoryValue.inputMethodology;
+      const methodology = findMethodology(
+        inventoryValue.inputMethodology as string,
+        gpcRefNo,
+      );
+      const activityKey = methodology?.activityTypeField as string;
+
+      dataDictionary[gpcRefNo as string] = {
+        // InventoryValue fields
+        inventory_year: inventoryYear,
+        gpc_reference_number: inventoryValue.gpcReferenceNumber,
+        notation_key: inventoryValue.unavailableReason,
+        input_methodology: inventoryValue.inputMethodology,
+        activityValues: activityValues.map((activityValue) => {
+          let activityTitleKey = activityValue.metadata?.activityTitle;
+          let dataQuality = activityValue.metadata?.dataQuality;
+          let dataSource = activityValue.metadata?.dataSource;
+          let activityAmount = activityValue.activityData?.[activityTitleKey];
+          let activityUnit = t(
+            activityValue.activityData?.[`${activityTitleKey}-unit`],
+          );
+          let emission_co2 = activityValue.gasValues.find(
+            (g) => g.gas === "CO2",
+          )?.emissionsFactor.emissionsPerActivity;
+          let emission_ch4 = activityValue.gasValues.find(
+            (g) => g.gas === "CH4",
+          )?.emissionsFactor.emissionsPerActivity;
+          let emission_n2o = activityValue.gasValues.find(
+            (g) => g.gas === "N2O",
+          )?.emissionsFactor.emissionsPerActivity;
+          return {
+            ghg_co2: activityValue.activityData?.co2_amount,
+            ghg_ch4: activityValue.activityData?.ch4_amount,
+            ghg_n2o: activityValue.activityData?.n2o_amount,
+            activity_type: t(activityValue.activityData?.[activityKey]),
+            emission_factor_unit: null,
+            emission_co2,
+            emission_ch4,
+            emission_n2o,
+            activity_amount: activityAmount,
+            activity_unit: activityUnit,
+            methodology: methodologyDescription,
+            activity_data_quality: dataQuality,
+            activity_data_source: dataSource,
+            total_co2e: toDecimal(activityValue.co2eq as bigint)
+              ?.div(new Decimal("1e3"))
+              .toNumber(),
+          };
+        }),
+      };
+    });
+    return dataDictionary;
+  }
+
+  private static replacePlaceholdersInRow(
+    row: Excel.Row,
+    dataSection: Record<string, any>,
+    rowNumber: number,
+    visitedScopes: Record<string, any>,
```

Comment:
Can you specify the types here instead of using any?

Reply:
TODO created